tox.ini

```diff
@@ -11,6 +11,11 @@ envlist = py27,
           docs
 minversion = 2.9.0
 skip_missing_interpreters = true
+isolated_build = true
+
+[testenv:.package]
+# no additional dependencies besides PEP 517 for building the package
+deps =
```

Comment:
This is only necessary because we are pinning to `tox<3.8.0`, right?

I modified `setup.py` to throw an exception if `pytest` or `six` is installed while building and on `3.7.0` it will fail, but on `3.21.1` it succeeds.

Considering that the downside of "the package is built with some additional stuff installed" is very low, maybe instead we should pin `tox` only on the `tz` job and leave this out?

Reply:
Yup, I can remove that once https://github.com/dateutil/dateutil/pull/1132 is landed.

Reply:
Right... but the tz environment still needs to run with tox<3.8.0, so it still needs that hack.

Reply:
I've added it back

Reply:
🙄 and it fails py34 with the new version of tox. I'm gonna leave the requirement and we can remove all this crap once we drop 3.4 altogehter.

Reply:
OK, SGTM.